docs/tutorials/configure/pet-photographer.md

```diff
@@ -0,0 +1,984 @@
+func (c \*colorFilterCam) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
+camStream, err := c.actualCam.Stream(ctx, errHandlers...)
+    if err != nil {
+        return nil, err
+    }
+
+    return filterStream{camStream, c.visionService}, nil
+
+}
+
+// NextPointCloud does nothing.
+func (c \*colorFilterCam)NextPointCloud(ctx context.Context)(pointcloud.PointCloud, error)
+{
+    return nil, errUnimplemented
+}
+
+// Properties returns details about the camera.
+func (c \*colorFilterCam) Properties(ctx context.Context)(camera.Properties, error)
+{
+    return c.actualCam.Properties(ctx)
+}
+
+// Projector does nothing.
+func (c \*colorFilterCam) Projector(ctx context.Context) (transform.Projector, error) {
+    return nil, errUnimplemented
+}
+
+type filterStream struct {
+    cameraStream gostream.VideoStream
+    visionService vision.Service
+}
+
+// Next contains the filtering logic and returns select data from the underlying camera.
+func (fs filterStream) Next(ctx context.Context) (image.Image, func(), error) {
+    if ctx.Value(data.FromDMContextKey{}) != true {
+        // If not data management collector, return underlying stream contents without filtering.
+        return fs.cameraStream.Next(ctx)
+    }
+
+    // Only return captured image if it contains a certain color set by the vision service.
+    img, release, err := fs.cameraStream.Next(ctx)
+    if err != nil {
+        return nil, nil, errors.New("could not get next source image")
+    }
+    detections, err := fs.visionService.Detections(ctx, img, map[string]interface{}{})
+    if err != nil {
+        return nil, nil, errors.New("could not get detections")
+    }
+
+    if len(detections) == 0 {
+        return nil, nil, data.ErrNoCaptureToStore
+    }
+
+    return img, release, err
+}
+
+// Close closes the stream.
+func (fs filterStream) Close(ctx context.Context) error {
+    return fs.cameraStream.Close(ctx)
+}
+```
+
+- A modular camera coded in Go looks for a flag called `fromDM` in the context (`ctx`) using `ctx.Value(data.FromDMContextKey{})` to figure out if data management triggered the filter, rather than using `extra`.
+
+  - Instead of implementing [`GetImage`](/components/camera/#getimage) (like it is in Python, etc), in Go, you will implement `Stream` as shown in the example code.
+
+- For other programming languages, similar utility functions will be exposed to help you check the caller of your filter function.
+  The approach to perform this check may differ depending on the particular function and programming language. For detailed information, please refer to your chosen language's SDK documentation.
+- If the boolean is `true`, the function will call the vision service to get detections and return the image if the color is detected; otherwise, it will raise the [`ErrNoCaptureToStore`](https://github.com/viamrobotics/rdk/blob/214879e147970a454f78035e938ea853fcd79f17/data/collector.go#L44) error.
```

Comment:
I seee, I included them because there was a desired emphasis on the ability to later create your own module with filtering capabilities. So I was trying to outline some of the details regarding the functionality that I could understand from the code.